BackendCommands.php

```diff
@@ -134,43 +118,162 @@ public function updateCode()
+     * TODO: folder anymore. Then decide if we can make this a post command for config:export.
+     *
      * @command backend:config-export
      *
      * @aliases backend:cex
      *
      * @options-backend
      *
-     * @validate-site-alias
-     *
-     * @usage drush @elle backend:export-config
+     * @usage drush @elle backend:config-export
      *   Exports the elle configuration.
      *
      * @bootstrap config
      */
     public function configExport()
     {
+        if ($this->environment !== 'prod') {
+            $this->logger()->error('Only production will be exported. Current environment is "%environment".', ['%environment' => $this->environment]);
+
+            return;
+        }
+
         // export the config into the export folder.
-        $this->drush($this->selfRecord(), 'config:export', [], ['yes' => true]);
+        $this->drush($this->selfRecord(), 'config:export', [], ['yes' => $this->input()->getOption('yes')]);
+    }
+
+    /**
+     * Move files from sync folder to shared or override folders.
+     *
+     * As long as we have to handle environment specific config, this can not
+     * be a post command for the default drush config:import command.
+     *
+     * TODO: Revisit when we do not need the local- and testing-environment config
+     * TODO: folder anymore. Then decide if we can make this a post command for config:export.
+     *
+     * @hook post-command backend:config-export
+     *
+     * @param $result
+     * @param \Consolidation\AnnotatedCommand\CommandData $commandData
+     */
+    public function postConfigExportCommand($result, CommandData $commandData)
+    {
+        $exportedFiles = $this->getConfigFilesInDirectory($this->siteConfigSyncDirectory());
+        $overrideFiles = $this->getConfigFilesInDirectory($this->siteConfigOverrideDirectory());
+        $sharedFiles = $this->getConfigFilesInDirectory($this->configSharedDirectory());
+        $localFiles = $this->getConfigFilesInDirectory($this->siteConfigEnvironmentDirectory('local'));
+        $modifiedFiles = [];
+
+        foreach ($exportedFiles as $fileName => $fullPath) {
+            // First check, if the file should be put into the override directory.
+            // otherwise check, if the file should be put into the shared directory.
+            if (isset($overrideFiles[$fileName])) {
+                if (!$this->filesAreEqual($overrideFiles[$fileName], $fullPath)) {
+                    $this->filesystem->copy($fullPath, $overrideFiles[$fileName], true);
+                    $modifiedFiles[$fileName] = $fullPath;
+                }
+            } elseif (isset($sharedFiles[$fileName])) {
+                if (!$this->filesAreEqual($sharedFiles[$fileName], $fullPath)) {
+                    $this->filesystem->copy($fullPath, $sharedFiles[$fileName], true);
+                    $modifiedFiles[$fileName] = $fullPath;
+                }
+            }
+        }
```

Comment:
Since you are adding .gitignore for configuration files in the 'sync' folder, see [backend merge request](https://tech-git.elle.de/burda_style/backend/-/merge_requests/231), files for newly created configuration exist only in 'sync' folder and therefore cannot be seen using git diff. I am undecided if they should be moved to shared or overridden.

Reply:
Furthermore, doing cim after cex will delete config.

Reply:
> 
> 
> Since you are adding .gitignore for configuration files in the 'sync' folder, see [backend merge request](https://tech-git.elle.de/burda_style/backend/-/merge_requests/231), files for newly created configuration exist only in 'sync' folder and therefore cannot be seen using git diff. I am undecided if they should be moved to shared or overridden.

My reasoning for doing it this way, is, that configuration that is not in git will definitely lead to something not working as expected. That would indicate to the developer, that he has missed something.
Silently moving it to one of the config folders would lead to a working codebase, but we would have silently made the decision, where the config should go. This might be a bigger problem.

Reply:
> 
> 
> Furthermore, doing cim after cex will delete config.

Oh, yes it will. Hm, we could add a "new config" folder, and move the new config there. But not use that folder on config import.
This way it would not get lost, but it would also lead to broken installations (which is wanted, so above)

Reply:
> My reasoning for doing it this way, is, that configuration that is not in git will definitely lead to something not working as expected

I am not so sure that this would be easy to realize. Say, if you are enabling a module which ships with some default config, you align that config, maybe some minor thing like a time period or whatever, export that config but do not move it, you could have the module still working on default config and it could possibly take you some time to realize what happened.